Shetland_PCPIP_Indicators.Rmd

```diff
@@ -588,3 +589,27 @@ pharma1_180_plotly
 pharma1_180_dt
 ```
 
+```{r export-all-data-frames, message=FALSE, warning=FALSE}
+# Create the named list of data frames and their sheet names
+all_data_frames <- list(
+  all_data = all_data,
+  monthly_event_type = monthly_event_type,
+  monthly_summary = monthly_summary,
+  pharma_1_monthly = pharma_1_monthly,
+  pharmacist_reviews = pharmacist_reviews,
+  quarterly_event_type = quarterly_event_type
+)
+
+# Create a new workbook
+wb <- createWorkbook()
+
+# Add each data frame as a sheet and set column widths to auto
+for (sheet_name in names(all_data_frames)) {
+  addWorksheet(wb, sheet_name)
+  writeData(wb, sheet_name, all_data_frames[[sheet_name]])
+  setColWidths(wb, sheet = sheet_name, cols = 1:ncol(all_data_frames[[sheet_name]]), widths = "auto")
```

Comment:
Look at formatting

- [ ] Column widths don't seem to be auto scaling :shrug:
- [ ] Bold the headers
- [ ] Check the `census_date` column - The indicators from `pharma.R` have 'times' on them. Perhaps an `as.Date()` is needed somewhere in the R code? Or get `openxlsx` to format all the' census_date' columns as a nice 'Excel Date'.
- [ ] Maybe rename columns - Go back into the original script, and use `CTRL + SHIFT + F` to find all uses to make sure you're not breaking things.
- [ ] Maybe make a table / add filters
- [ ] Maybe format columns as percentages (might be v tricky).